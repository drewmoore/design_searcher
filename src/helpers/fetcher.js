import fetch from 'isomorphic-fetch';

// Module for wrapping isomorphic-fetch. This allows stubbing in tests.
export default class Fetcher {
  static fetch(url, options) {
    return fetch(url, options);
  }
};
