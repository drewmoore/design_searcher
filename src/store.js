import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import root  from './reducers/root';

const middleWare = applyMiddleware(thunk);
const store      = createStore(root, {}, middleWare);

export default store;
