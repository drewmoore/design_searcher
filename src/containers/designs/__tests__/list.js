import React                       from 'react';
import { Provider }                from 'react-redux';
import renderer                    from 'react-test-renderer';
import { createStore }             from 'redux';
import { mount }                   from 'enzyme';
import { stubFetch, restoreFetch } from '../../../test_support/helpers';
import store                       from '../../../store';
import DesignsListContainer        from '../list';

describe('DesignsContainer', () => {
  let template;

  beforeEach(() => {
    stubFetch();
    template = (
      <Provider store={store}>
        <DesignsListContainer />
      </Provider>
    );
  });

  afterEach(() => restoreFetch());

  describe('DOM Snapshot', () => {
    test('matches snapshot', () => {
      const tree = renderer.create(template).toJSON();
      expect(tree).toMatchSnapshot();
    })
  });

  describe('Rendering Props from State', () => {
    let designs, component;

    beforeEach(() => {
      designs = store.getState().designs;
      component = mount(template);
    });

    test('renders names', () => (
      designs.items.forEach(design => (
        expect(component.text()).toMatch(design.name)
      ))
    ));

    test('renders thumbnail images', () => (
      designs.items.forEach(design => (
        expect(
          component.find(`img[src="${design.thumbnail_url}"]`).length
        ).toBeGreaterThan(0)
      ))
    ));

    test('renders descriptions if available', () => (
      designs.items.forEach(design => {
        if (design.short_description) {
          expect(component.text()).toMatch(design.short_description)
        }
      })
    ));

    test('renders number of favorites for a design', () => (
      designs.items.forEach(design => (
        expect(
          component.find(`[data-num-favorites=${design.num_favorites}]`).length
        ).toBeGreaterThan(0)
      ))
    ));

    test('renders purchase links based on design url slug', () => (
      designs.items.forEach(design => (
        expect(component.html()).toMatch(design.slug)
      ))
    ));
  });
});
