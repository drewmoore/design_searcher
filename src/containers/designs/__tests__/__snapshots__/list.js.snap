// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesignsContainer DOM Snapshot matches snapshot 1`] = `
<div
  className="row pad-top"
>
  <div
    className="col-xs-12 well text-center"
  >
    <strong>
      Please wait while we find designs ...
    </strong>
  </div>
</div>
`;
